### PR TITLE
TestBuild: Disable telemetry for test builds

### PR DESCRIPTION
**Pull request by kasperpeulen**
## What I did

Disable telemetry for test builds

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Review by ndelangen — APPROVED** [review]

File: code/lib/core-server/src/withTelemetry.ts [hunk 1]
```diff
@@ -120,11 +120,13 @@ export async function withTelemetry<T>(
   options: TelemetryOptions,
   run: () => Promise<T>
 ): Promise<T | undefined> {
+  const enableTelemtry = !(options.cliOptions.disableTelemetry || options.cliOptions.test === true);
```
Comment (shilman) [inline code/lib/core-server/src/withTelemetry.ts:123]:
```suggestion
  const enableTelemetry = !(options.cliOptions.disableTelemetry || options.cliOptions.test === true);
```

**Review by shilman — APPROVED** [review]
Let's do this for now, but hopefully figure out how to get these events without slowing down the build as a followup task later

File: code/lib/core-server/src/withTelemetry.ts [hunk 2]
```diff
@@ -120,11 +120,13 @@ export async function withTelemetry<T>(
...
+
   let canceled = false;
 
   async function cancelTelemetry() {
     canceled = true;
-    if (!options.cliOptions.disableTelemetry) {
+    if (enableTelemtry) {
```
Comment (ndelangen) [inline code/lib/core-server/src/withTelemetry.ts:129]:
```suggestion
    if (enableTelemetry) {
```

File: code/lib/core-server/src/withTelemetry.ts [hunk 3]
```diff
@@ -136,8 +138,7 @@ export async function withTelemetry<T>(
     process.on('SIGINT', cancelTelemetry);
   }
 
-  if (!options.cliOptions.disableTelemetry)
-    telemetry('boot', { eventType }, { stripMetadata: true });
+  if (enableTelemtry) telemetry('boot', { eventType }, { stripMetadata: true });
```
Comment (ndelangen) [inline code/lib/core-server/src/withTelemetry.ts:141]:
```suggestion
  if (enableTelemetry) telemetry('boot', { eventType }, { stripMetadata: true });
```

File: code/lib/core-server/src/withTelemetry.ts [hunk 4]
```diff
@@ -148,7 +149,8 @@ export async function withTelemetry<T>(
 
     const { printError = logger.error } = options;
     printError(error);
-    await sendTelemetryError(error, eventType, options);
+
+    if (enableTelemtry) await sendTelemetryError(error, eventType, options);
```
Comment (ndelangen) [inline code/lib/core-server/src/withTelemetry.ts:153]:
```suggestion
    if (enableTelemetry) await sendTelemetryError(error, eventType, options);
```

**Pull request merged**